xbmc/network/Socket.cpp

```diff
@@ -171,6 +138,42 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
   return true;
 }
 
+bool CPosixUDPSocket::CheckIPv6(int port, int range)
+{
+  bool ipv6 = false;
+  CAddress testaddr("::");
+
+  SOCKET testSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+  if (testSocket == INVALID_SOCKET)
+    return false;
+
+#ifdef WINSOCK_VERSION
+  const char zero = 0;
+#else
+  int zero = 0;
+#endif
+
+  if (setsockopt(testSocket, IPPROTO_IPV6, IPV6_V6ONLY, &zero, sizeof(zero)) == -1)
+  {
+    closesocket(testSocket);
+    return false;
+  }
+
+  // Try to bind a socket to validate ipv6 status
+  for (; port <= port + range; port++)
+  {
+    testaddr.saddr.saddr6.sin6_port = htons(port);
+    if (bind(testSocket, reinterpret_cast<struct sockaddr*>(&testaddr.saddr), testaddr.size) >= 0)
```

Comment:
bind should return 0 in success case, therefore the ``` >=``` is not needed. Also on failure this sets ```errno``` - just if we want to check that.